RecoLocalTracker/SiStripZeroSuppression/plugins/SiStripHybridFormatAnalyzer.cc

```diff
@@ -0,0 +1,283 @@
+      std::vector<int> pedestals;
+      uint32_t peds_cache_id;
+	  
+	  
+};
+
+
+SiStripHybridFormatAnalyzer::SiStripHybridFormatAnalyzer(const edm::ParameterSet& conf){
+   
+
+  srcDigis_ =  conf.getParameter<edm::InputTag>( "srcDigis" );
+  srcAPVCM_ =  conf.getParameter<edm::InputTag>( "srcAPVCM" );
+  nModuletoDisplay_ = conf.getParameter<uint32_t>( "nModuletoDisplay" );
+  plotAPVCM_ = conf.getParameter<bool>( "plotAPVCM" );
+
+  sdDigis_= fs_->mkdir("Digis");
+  sdMisc_= fs_->mkdir("Miscellanea");
+  
+  h1APVCM_ = sdMisc_.make<TH1F>("APV CM","APV CM", 1601, -100.5, 1500.5);
+  h1APVCM_->SetXTitle("APV CM [adc]");
+  h1APVCM_->SetYTitle("Entries");
+  h1APVCM_->SetLineWidth(2);
+  h1APVCM_->SetLineStyle(2);
+  
+   h1BadAPVperEvent_ = sdMisc_.make<TH1F>("BadAPV/Event","BadAPV/Event", 72786, -0.5, 72785.5);
+   h1BadAPVperEvent_->SetXTitle("# Bad APVs");
+   h1BadAPVperEvent_->SetYTitle("Entries");
+   h1BadAPVperEvent_->SetLineWidth(2);
+   h1BadAPVperEvent_->SetLineStyle(2);
+  	
+   h1BadAPVperModule_ = sdMisc_.make<TH1F>("BadAPV/Module","BadAPV/Module", 7, -0.5, 6.5);
+   h1BadAPVperModule_->SetXTitle("# Bad APVs");
+   h1BadAPVperModule_->SetYTitle("Entries");
+   h1BadAPVperModule_->SetLineWidth(2);
+   h1BadAPVperModule_->SetLineStyle(2);
+   
+   h1BadAPVperModuleOnlyBadModule_ = sdMisc_.make<TH1F>("BadAPV/Module Only Bad Modules","BadAPV/Module Only Bad Modules", 7, -0.5, 6.5);
+   h1BadAPVperModuleOnlyBadModule_->SetXTitle("# Bad APVs");
+   h1BadAPVperModuleOnlyBadModule_->SetYTitle("Entries");
+   h1BadAPVperModuleOnlyBadModule_->SetLineWidth(2);
+   h1BadAPVperModuleOnlyBadModule_->SetLineStyle(2);
+   
+   h1Pedestals_ = sdMisc_.make<TH1F>("Pedestals","Pedestals", 2048, -1023.5, 1023.5);
+   h1Pedestals_->SetXTitle("Pedestals [adc]");
+   h1Pedestals_->SetYTitle("Entries");
+   h1Pedestals_->SetLineWidth(2);
+   h1Pedestals_->SetLineStyle(2);  
+     
+}
+
+
+SiStripHybridFormatAnalyzer::~SiStripHybridFormatAnalyzer()
+{
+ 
+   
+
+}
+
+void
+SiStripHybridFormatAnalyzer::analyze(const edm::Event& e, const edm::EventSetup& es)
+{
+   using namespace edm;
+
+     //plotting pedestals
+     //------------------------------------------------------------------
+     if(actualModule_ ==0){
+      	uint32_t p_cache_id = es.get<SiStripPedestalsRcd>().cacheIdentifier();
+      	if(p_cache_id != peds_cache_id) {
+			es.get<SiStripPedestalsRcd>().get(pedestalsHandle);
+			peds_cache_id = p_cache_id;
+      	}
+      	std::vector<uint32_t> detIdV;
+      	pedestalsHandle->getDetIds(detIdV);
+      
+      	for(uint32_t i=0; i < detIdV.size(); ++i){
+        	pedestals.clear();
+        	SiStripPedestals::Range pedestalsRange = pedestalsHandle->getRange(detIdV[i]);
+        	pedestals.resize((pedestalsRange.second- pedestalsRange.first)*8/10);
+			pedestalsHandle->allPeds(pedestals, pedestalsRange);
+			for(uint32_t it=0; it < pedestals.size(); ++it) h1Pedestals_->Fill(pedestals[it]);
```

Comment:
That's also how I understand it: this will fill the pedestals into a histogram for all modules up until the first one that has a non-empty `SiStripDigi` set. The caching is to avoid repeatedly getting the `SiStripPedestals` from the event setup. I reordered this a bit (making all except the `peds_cache_id_` variable local and using range-based for loops), but I don't know what the intention is. @icali @CesarBernardes could you clarify? In any case this code is only used for making plots during development.

Reply:
if this code is moved to the /test directory, there is more freedom to have somewhat questionable implementation and little restrictions from the coding style point of view.